lib/src/json_serializable/type_helper.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+abstract class TypeHelper {
+  const TypeHelper();
+
+  bool matchesTypeForSerialize(DartType type);
+
+  bool canDeserialize(DartType type);
+
+  String toSerializedValue(DartType searchType, String name);
```

Comment:
Lets add comments to these methods, at least toSerializedValue and toDeserializedValue. I was confused why they took Strings and returned Strings until I got to the implementations and realized they were returning actual code to be written out as a String.